src/headers_ext/mod.rs

```diff
@@ -0,0 +1,16 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+// This file is part of Static Web Server.
+// See https://static-web-server.net/ for more information
+// Copyright (C) 2019-present Jose Quintana <joseluisq.net>
+
+//! Additional types for the headers module in order to handle Accept-Encoding
+//! header.
+//!
+
+mod accept_encoding;
+mod content_coding;
+mod quality_value;
+
+pub use accept_encoding::AcceptEncoding;
+pub use content_coding::ContentCoding;
+pub use quality_value::QualityValue;
```

Comment:
Let's also make those three modules and their methods internal.
Likely some other methods will have to be removed as a result.

Reply:
I made all of `headers_ext` internal. Yes, some unused code had to be removed, but also code examples – doc tests don’t work with private modules. And there have been more clippy warnings, so I’ve renamed `to_static` into `as_str` to match conventions. There was also a suggestion to rename `ContentCoding::GZIP` into `ContentCoding::Gzip`, a warning I silenced for now.